content/posts/redis-modules-into-arm-land-part1.md

```diff
@@ -0,0 +1,275 @@
+and again that newer systems work better and old ones can't keep up.
+
+Installing an OS brings us to the first dilemma: 
+If we install Raspbian, we get a 32-bit OS (i.e., arm32v7 platform).
+If we choose 64-bit Ubuntu, we get an arm64v8 platform (we discuss ARM platforms in detail later). 
+If we intend (as we do with RedisEdge) to support both platforms, we can either get two SD cards, and install each OS on its own card, taking turns on the device, or get two RPi devices (which is not terribly expensive). I recommend the latter.
+
+And now, for the principal principle of OS selection: our goal is to have a stable system with the newest Docker version. Any OS that will satisfy these requirements will do.
+That's right: we're going to use Docker to fetch the OS we really with to build for, using the underlying OS as infrastructure. This will also help keep our build experiments properly isolated.
+
+If you have a RPi device at your disposal, you can now proceed to making it functional. Otherwise, you can skip this section, as we will present build methods that do not require a physical ARM machine.
+
+#### Installing Ubuntu 19.04 on RPi 3 & 4
+
+Very good and detailed instructions of how to download and install the latest Ubuntu Server for ARM on RPi can be found [here](https://ubuntu.com/download/iot/raspberry-pi), for Linux, Windows, and macOS. However, you should follow the following instructions to install Ubuntu 19.04 rather than the latest released version:
+
+- Get a microSD Card: 64GB, Class 10, anything comparable (that is, at the same price level) to Samsung EVO or SanDisk Ultra will do.
+- Download the OS [image](http://cdimage.ubuntu.com/releases/19.04/release/ubuntu-19.04-preinstalled-server-arm64+raspi3.img.xz) (typically .img or .raw format, archived) and extract it. While it is possible to install from an .iso file, using .img images is much more straightforward and therefore recommended.
+- Write it to a microSD card with a tool like [Balena Etcher](https://www.balena.io/etcher/) (it's available for all platforms, I personally use [Win32 Disk Imager](https://sourceforge.net/projects/win32diskimager/)). In order to write, you should use the build-in microSD reader in your laptop or get an external one with a USB interface.
+- Insert the microSD into the RPi and power the device on.
+- Username/password: ubuntu/ubuntu
+
+#### Installing Raspbian Buster on RPi 3 & 4
+
+* Repeat the above steps, with Raspbian Lite [image](https://downloads.raspberrypi.org/raspbian_lite_latest).
+* Username/password: pi/raspberry
+
+#### Connecting to Workstation
+
+By "workstation" I refer to a Linux or macOS host, that holds one's development environment and git repositories. As a side note,  I warmly recommend using a desktop PC (that's another blog post), though most people use laptops. In either case, I also recommend having some virtualization infrastructure on your workstation, VMware Workstation/Fusion or VirtualBox being good solutions.

```

Comment:
Change to: By "workstation," I'm referring to a Linux or macOS host that holds one's development environment and git repositories. As a side note, I highly recommend using a desktop PC (that's another blog post), though most people use laptops. In either case, I also recommend having some virtualization infrastructure on your workstation, such as VMware Workstation/Fusion or VirtualBox.